lib-injection/host_inject.rb

```diff
@@ -0,0 +1,97 @@
+return if ENV["DD_TRACE_SKIP_LIB_INJECTION"] == "true"
+
+Gem.paths = {
+  "GEM_PATH" => "/opt/datadog/apm/library/ruby/#{RUBY_VERSION.split(".")[0..1].join(".")}:/opt/datadog/apm/library/ruby:#{ENV["GEM_PATH"]}"
+}
+
+ENV["GEM_PATH"] = Gem.path.join(":")
```

Comment:
This is a hard change to the user's environment variable, is this intended like this?
If so, can we document what exactly we want to accomplish in this line? I ask because overriding the GEM_PATH will have serious side effects, so other might question this same line in the future.